### PR TITLE
Issue1209 thermal zone moist air

### LINKED ISSUE #1209: Enhance ROM (ThermalZone.RC) for moisture balance
**What do I want to do?**
I want to implement the possibility to take into account moisture production and moisture balance in thermal zones. 
For this reason I would like to modify the air volume model in the `ThermalZones.ReduceOrder.RC.OneElement.mo`.

**How I want to solve it**

- I will implement a boolean parameter `use_moisture_balance` that will be _false_ per default.

- The `volAir` will be conditioned with this parameter. If it is _false_ the existing `volAir` will be used.

- A second air volume `volMoistAir` will be implemented (type: `Fluid.MixingVolumes.MixingVolumeMoistAir`) and conditioned by the same parameter (if _true_ use `volMoistAir`)

- An additional input connector `mWat_flow` and output connector `X_w` both conditioned with `use_moisture_balance` will be implemented and conditionally connected to the `volMoistAir`

In the Aixlib this implementation has been performed. To avoid Merge-Conflicts with the IBPSA, I would like to implement the same changes into the IBPSA. Changes can be seen in issue [695](https://github.com/RWTH-EBC/AixLib/issues/695) of the Aixlib.

Are there any concerns about this changes?

### PR DESCRIPTION
**Pull request by mwetter**
This closes #1209.

### CODE REVIEW

**Comment by mwetter** [comment]
@Mathadon : Can you please do a last review and then merge.

File: IBPSA/ThermalZones/ReducedOrder/RC/OneElement.mo [hunk 1]
```diff
@@ -129,8 +139,25 @@ model OneElement "Thermal Zone with one element for exterior walls"
     final C_start=C_start,
     final C_nominal=C_nominal,
     final mSenFac=mSenFac,
-    final use_C_flow=false) if VAir > 0 "Indoor air volume"
-    annotation (Placement(transformation(extent={{38,-10},{18,10}})));
+    final use_C_flow=false) if VAir > 0 and not use_moisture_balance
+    "Indoor air volume"
+    annotation (Placement(transformation(extent={{42,-26},{22,-6}})));
+  Fluid.MixingVolumes.MixingVolumeMoistAir volMoiAir(
```
Comment (Mathadon) [inline IBPSA/ThermalZones/ReducedOrder/RC/OneElement.mo]: This makes me wonder why we have a separate model for `MixingVolume` with and without moisture..?

File: IBPSA/ThermalZones/ReducedOrder/RC/OneElement.mo [hunk 2]
```diff
@@ -233,12 +262,29 @@ protected
...
     annotation (Placement(transformation(extent={{-186,118},{-174,130}})));
 
+  Modelica.Blocks.Math.Gain mWat_flow(
+    final k(unit="kg/J") = 1/h_fg,
+    u(final unit="W"),
+    y(final unit="kg/s")) if
+       use_moisture_balance and ATot >0 "Water flow rate due to latent heat gain"
+    annotation (Placement(transformation(extent={{-200,-100},{-180,-80}})));
+
+  Buildings.HeatTransfer.Sources.PrescribedHeatFlow conQLat_flow if
```
Comment (Mathadon) [inline IBPSA/ThermalZones/ReducedOrder/RC/OneElement.mo]: needs IBPSA counterpart?

**Comment by Mathadon** [comment]
Indeed, unit tests fail. In case this is of interest, we can add 
```
after_script:
  - cat IBPSA/failed-simulator-dymola.log
```
to `.travis.yml`, which would allow to debug the CI output ` Simulation failed for 'RunScript("Resources/Scripts/Dymola/ThermalZones/ReducedOrder/Validation/RoomWithLatentGain.mos");'` etc more easily.

**Comment by Mathadon** [comment]
Btw, I'd be in favor of adding `Buildings.HeatTransfer.Sources.PrescribedHeatFlow` to IBPSA since it has better algebraic-loop-ish properties.

File: IBPSA/ThermalZones/ReducedOrder/RC/OneElement.mo (hunk 1 above)
Comment (mwetter) [reply IBPSA/ThermalZones/ReducedOrder/RC/OneElement.mo]: I wonder too.

File: IBPSA/ThermalZones/ReducedOrder/RC/OneElement.mo (hunk 2 above)
Comment (mwetter) [reply IBPSA/ThermalZones/ReducedOrder/RC/OneElement.mo]: Yes. Let's let travis fail, then I can add the `after_script` to see if it does the right thing, and then I can fix it.

**Comment by mwetter** [comment]
> Btw, I'd be in favor of adding `Buildings.HeatTransfer.Sources.PrescribedHeatFlow` to IBPSA since it has better algebraic-loop-ish properties. In which case it would make sense to replace its body by
> 
> ```
> extends Modelica.Thermal.HeatTransfer.Sources.PrescribedHeatFlow(final alpha=0)
> ```
> 
> to be a bit more didactic than the current implementation. Although it would require more translation overhead..

We could add `Buildings.HeatTransfer.Sources` to `IBPSA` if there is a consensus.

**Pull request merged**